app/src/main/java/com/github/multimatum_team/multimatum/activity/DeadlineDetailsActivity.kt

```diff
@@ -367,8 +398,9 @@ class DeadlineDetailsActivity : AppCompatActivity() {
      */
     private fun textNotification(): String {
         val notifications =
-            nameCheckBox.filter { t -> notificationSelected[nameCheckBox.indexOf(t)] }
+            nameNotifications.filter { t -> notificationSelected[nameNotifications.indexOf(t)] }
```

Comment:
Here too you should be able to use `filterIndexed` instead of this ugly `indexOf` :)